### PR TITLE
Release 0.2.0

### PR DESCRIPTION
**Pull request by Levia**
### Release 0.2.0

* Add realms information
* Add download functionality

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -5,24 +5,37 @@
...
   filename = "splits_with_attributes/out_#{file_index}.csv.gz"
   @rows = []
 
   FileManager.with_file(filename) do
-    species = S3.new.get_data(filename, params[:id])
+    species = S3.new.get_data(filename, @id)
+    @realms_counts = Species.count_by_realm(species)
     @total_count = species.count
-    @species = species.group_by { |hash| hash['category'] }
+    @species = species.group_by { |s| s.category }
```
Comment (lucacug) [inline app.rb]: you can probably get rid of the block here `species.group_by(&:category)`

File: lib/realms.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  REALMS = %w(biome_marine biome_terrestrial biome_freshwater).freeze
+
+  def realm_names
+    REALMS.map { |r| r.split('_').last }
+  end
+
+  def get_realms(s)
+    realm_names.map do |r|
+       is_true?(s.send("biome_#{r}")) ? r.capitalize : nil
```
Comment (lucacug) [inline lib/realms.rb]: fix indentation

File: lib/s3.rb [hunk 3]
```diff
@@ -34,10 +34,10 @@ def get_data(filename, cell_id)
         # :stats has statistical information about the data transferred such as:
         # => Aws::S3::Types::Stats bytes_scanned=xx, bytes_processed=xx, bytes_returned=xx
         # :end indicates the end of the event
-        @species = CSV.parse(event.payload.read).map {|a| Hash[ @keys.zip(a) ] } if is_records_event?(event)
+        @species = CSV.parse(event.payload.read).map {|a| Species.new(Hash[ @keys.zip(a) ]) } if is_records_event?(event)
```
Comment (lucacug) [inline lib/s3.rb]: I would leave space within the curly braces and remove it from inside the square ones

File: lib/species.rb [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    'CR' => 'Critically Endangered',
+    'EN' => 'Endangered',
+    'VU' => 'Vulnerable',
+    'NT' => 'Near Threatened',
+    'DD' => 'Data Deficient'
+  }.freeze
+  THREATENED = %w(CR EN VU).freeze
+
+  # Make KEYS plus :realms accessible instance attributes
+  attr_accessor *KEYS.map(&:downcase).map(&:to_sym) << :realms
```
Comment (lucacug) [inline lib/species.rb:24]: Why did you use the splat operator here?

File: lib/species.rb [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    species.each do |s|
+      s.realms = Realms.get_realms(s)
+    end
+    species
+  end
+
+  def self.count_by_realm(species)
+    counts = {}
+    multiple = 0
+    _species = species.group_by { |s| s.realms }
```
Comment (lucacug) [inline lib/species.rb]: I would use  `&:realms` rather than the block here again.

File: lib/species.rb [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+    _species.each do |realm, list|
+      if realm.include?('/')
+        multiple +=  list.count
```
Comment (lucacug) [inline lib/species.rb]: extra empty space, sorry.

**Review by lucacug — COMMENTED** [review]
@Levia I left a few comments mostly style-related. Tested it on my local machine and the results match the one on staging, so good to merge once comments addressed.

File: lib/species.rb (hunk 4 above)
Comment (Levia) [reply lib/species.rb:24]: It was necessary as `attr_accessor` doesn't accept an array as parameter

**Comment by Levia** [comment]
Thanks @lucacug. I'll give it a go at fixing those soon

**Pull request merged**